dlib/simd/simd8f_vec.h

```diff
@@ -0,0 +1,251 @@
+// Copyright (C) 2013  Davis E. King (davis@dlib.net)
+// License: Boost Software License   See LICENSE.txt for the full license.
+#ifndef DLIB_sIMD8F_Hh_
```

Comment:
Yes,  and I wanted you to look at tell me what you wanted done with these files.
I can merge the 4i/4f_vec  into the regular 4i/4f  like   C/SSE/AVX code if you like.   I think it's cleaner separate if there is going to be SSE2/4,  AVX,  NEON, VSX.   There are a few things that could be done however.   

 Instead of adding per  extension all mingled,   they could be separated like the classes are.   and each area will have the all function definitions large defined area.

Each specific ARCH/ext  could use its own header like _vec

Or separate .cpp files,  (inlining can still be used,  it's applied at code generation level with optimizations IIRC)

The last option is the cleanest,  but would require some modification to the cmake build.





#elif   AVX
#elif  ....
}